### DISCUSSION TITLE
disable katex equation automatically resizing

### DISCUSSION THREAD

**Discussion by ChiMaoShuPhy** [marp-team/marp#159] [Q&A] [answered]
Thanks for your always kind and quick assistance!

I am using the marp vscode extension

I want to ask is there an option to disable katex equation being resized when the equation is very wide? In stead, one may prefer overflow-x scroll.

besides, the katex equation (not exceeding the width of the section) is also resized in custom properties, for an example

I defined a custom property `embox` in my custom theme css 

```css

embox {
    background-color: #cfc;
    display:inline-block;
    width: fit-content;
    max-width: 100%;
}
```

in the marp mark down, i have a wide equation

```
* test in `embox`

  <embox>

    $$
    f(\relax{x}) = \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi
    $$

  </embox>

* test without `embox`
    $$
    f(\relax{x}) = \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi
    $$
```

what I got is

![image](https://user-images.githubusercontent.com/16471347/132107076-75b334c6-5230-40e3-ba8e-5b828c35d08f.png)

I have to manually set the width of `embox` in style. I am wondering is there any way to prevent katex from auto resizing and adjust the width of `embox` according to the original width or katex equation.

Many thanks!

**Comment by yhatt** [comment, marked as answer]
> I want to ask is there an option to disable katex equation being resized when the equation is very wide? In stead, one may prefer overflow-x scroll.

KaTeX auto-scaling is enabled by Marp Core theme.
https://github.com/marp-team/marp-core/tree/main/themes#auto-scaling-flags

You have to use a custom theme if you want to disable KaTeX auto-scaling. Exclude `math` from the scaling targets by using `@auto-scaling` metadata.

```css
/* @theme custom-theme */
/* @auto-scaling fittingHeader,code */

@import 'default';
```

> :warning: Please note that [any `overflow` CSS declarations cannot work on PDF](https://stackoverflow.com/questions/44660044/markdown-to-pdf-and-scrollbars#comment76309325_44660044). Marp is prioritizing that there are no problems in PDF export by default.

**Reply by ChiMaoShuPhy** [reply]
@yhatt  Somehow this issue appears again after I reinstalled the latest marp tools. I have tested with the same example as I posted in the question. I have checked that I set `math: katex` in the constructor options and exclude `math` from `@auto-scaling` meta data in the custom css theme file, but the katex element is still resized as the snapshot in the question

The css theme file:
```css
/* @theme my_theme */
/* @auto-scaling fittingHeader */

@import 'default';

embox {
    background-color: #cfc;
    display:inline-block;
    width: fit-content;
    max-width: 100%;
}
```
The mark down file:
```markdown
---
marp: true
theme: my_theme
_class: lead
backgroundColor: #fff
paginate: true
size: 4:3
math: katex 

---

* test in `embox`

  <embox>

    $$
    f(\relax{x}) = \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi
    $$

  </embox>

* test without `embox`
    $$
    f(\relax{x}) = \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi \int_{-\infty}^\infty   f(\hat\xi)\,e^{2 \pi i \xi x}   \,d\xi
    $$
```

**Reply by yhatt** [reply]
Thank you for reporting. This may be a regression of Marp Core that has been overlooked over the years since the release of v3.0.0, which [introduced drastic changes to the auto-scaling rendering](https://marp.app/blog/202205-ecosystem-update#renewed-auto-scaling-component).

Please track the progress status in https://github.com/marp-team/marp-core/issues/353.

**Reply by yhatt** [reply]
Fixed in [Marp Core v3.8.1](https://github.com/marp-team/marp-core/releases/tag/v3.8.1).

If you were using [Marp CLI in your Node.js project](https://github.com/marp-team/marp-cli#local-installation), [you can test the latest core with `npm i @marp-team/marp-core@^3.8.1 --save-dev`](https://github.com/marp-team/marp-cli#use-specific-version-of-marp-core). If not, please be patient for updation of downstream Marp ecosystem tools. :)
